src/private/MDMPerformerGroup.h

```diff
@@ -37,7 +37,7 @@
 // nil by default. Useful for view duplication.
 @property(nonatomic, nullable) id schedulerTarget;
 
-- (void)addPlan:(nonnull id<MDMPlan>)plan trace:(nonnull MDMTrace *)trace;
+- (void)addPlan:(nonnull id<MDMPlan>)plan trace:(nonnull MDMTrace *)trace log:(nullable MDMTransactionLog *)log;
```

Comment:
You may find this implementation to be cleaner to implement if you break it out to match the public APIs of addPlan, addPlanNamed, and removePlanNamed and then drop use of the transaction log object. Had you considered this already?